### PR TITLE
allow signatures with DSA subkeys

### PR DESCRIPTION
**Pull request by maxtaco**
- also fix a boolean formula bug in the computation of the fallback primary key

### CODE REVIEW

File: openpgp/keys.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func (e *Entity) encryptionKey(now time.Time) (Key, bool) {
 		// so this sort of thing is pretty important for encrypting to older keys.
 		//
 		if ((subkey.Sig.FlagsValid && subkey.Sig.FlagEncryptCommunications) ||
-		    (!subkey.Sig.FlagsValid && subkey.PublicKey.PubKeyAlgo == packet.PubKeyAlgoElGamal)) &&
+			(!subkey.Sig.FlagsValid && subkey.PublicKey.PubKeyAlgo == packet.PubKeyAlgoElGamal)) &&
```
Comment (maxtaco) [inline openpgp/keys.go:111]: whoops forgot to format with the last release

File: openpgp/keys.go [hunk 2]
```diff
@@ -164,8 +163,10 @@ func (e *Entity) signingKey(now time.Time) (Key, bool) {
 	// If we have no candidate subkey then we assume that it's ok to sign
 	// with the primary key.
 	i := e.primaryIdentity()
-	if (!i.SelfSignature.FlagsValid || i.SelfSignature.FlagSign &&
-		!i.SelfSignature.KeyExpired(now)) && e.PrivateKey.PrivateKey != nil {
+	if (!i.SelfSignature.FlagsValid || i.SelfSignature.FlagSign) &&
+		e.PrimaryKey.PubKeyAlgo.CanSign() &&
+		!i.SelfSignature.KeyExpired(now) &&
+		e.PrivateKey.PrivateKey != nil {
```
Comment (maxtaco) [inline openpgp/keys.go:169]: same logic in both places.  should indeed be factored out into a helper function.

**Comment by oconnor663** [comment]
:ship:

**Pull request merged**